### DISCUSSION TITLE
Integrating two different radios

### DISCUSSION THREAD

**Discussion by williamzhang0306** [nasa/fprime#2675] [Q&A]
Hello! I'm working on a high level software design in Fprime for a cubesat my university is developing.

On our cubesat we will have a UHF transceiver and an S-band transmitter.  Our plan is to use UHF for up/downlink of commands and telemetry and S-band just for downlink of the larger files the cubesat will be generating.

I've read through the documentation on [ground interfaces](https://nasa.github.io/fprime/UsersGuide/best/ground-interface.html) and found that you could use different drivers for uplink vs downlink. However, I'm not sure how we would implement different drivers for downlink dependent on the type of data. I would appreciate any advice!

**Comment by LeStarch** [comment]
You could have two separate sets of ground downlink components. One for UHF and the other for S-band.  Then hookup the correct components for each!

**Comment by ReggieMarr** [comment]
@williamzhang0306 I'm actually working on a project very similar to this and I'd be curious to see what kind of hardware stack you're working with.

I've been developing an fprime plugin in the background to allow for hot swapping between a udp interface and s-band transmitter which could be pretty relevant.
